yellowbrick/classifier/base.py

```diff
@@ -157,18 +201,131 @@ def score(self, X, y, **kwargs):
         ----------
         X : array-like
             X (also X_test) are the dependent variables of test set to predict
+
         y : array-like
             y (also y_test) is the independent actual variables to score against
 
         Returns
         -------
         score : float
+            Returns the score of the underlying model, usually accuracy for
+            classification models. Refer to the specific model for more details.
         """
-        self.score_ = self.estimator.score(X, y, **kwargs)
+        # If the estimator has been passed in fitted but the visualizer was not fit
+        # then we can retrieve the classes from the estimator, unfortunately we cannot
+        # retrieve the class counts so we simply set them to None and warn the user.
+        # NOTE: cannot test if hasattr(self, "classes_") because it will be proxied.
+        if not hasattr(self, "class_counts_"):
+            if not hasattr(self.estimator, "classes_"):
+                raise NotFitted(
+                    (
+                        "could not determine required property classes_; "
+                        "the visualizer must either be fit or instantiated with a "
+                        "fitted classifier before calling score()"
+                    )
+                )
+
+            self.class_counts_ = None
+            self.classes_ = self._decode_labels(self.estimator.classes_)
+            warnings.warn(
+                "could not determine class_counts_ from previously fitted classifier",
+                YellowbrickWarning,
+            )
 
+        # This method implements ScoreVisualizer (do not call super).
+        self.score_ = self.estimator.score(X, y)
         return self.score_
 
-    # TODO during refactoring this can be used to generalize ClassBalance
-    def class_counts(self, y):
-        unique, counts = np.unique(y, return_counts=True)
-        return dict(zip(unique, counts))
+    def _decode_labels(self, y):
+        """
+        An internal helper function that uses either the classes or encoder
+        properties to correctly decode y as user-readable string labels.
+
+        If both classes and encoder are set, a warning is issued and encoder is
+        used instead of classes. If neither encoder nor classes is set then the
+        original array is returned unmodified.
+        """
```

Comment:
This raises an error if y is already decoded -- but that seems like a weird situation and the errors raised are clear about what's gone wrong. Let me know if we need to handle this scenario.

Reply:
I agree, this seems like an awfully tangled situation that the user would have to get themselves into, and the errors should be clear enough in that unlikely situation!